frontend/src/metabase/components/ArchiveModal.jsx

```diff
@@ -19,7 +19,7 @@ class ArchiveModal extends React.Component {
       // Note - if using this component inside of a modal route, make sure onClose is called in that component and not here, otherwise any attmempts to push to a new location won't work properly
       // Currently the only place we don't use this in a modal route is archiving a question
       if (onClose) {
-        onClose();
+        onClose(true);
```

Comment:
This won't work because `onClose` is also called when cancelling the modal https://github.com/metabase/metabase/blob/e752c2874fb241361460c093dab1c95c2bf4f27d/frontend/src/metabase/components/ArchiveModal.jsx#L37

Reply:
In this case we want specific behavior when closing as a result of archiving. So when the close button is clicked we actually don't want to pass anything